packages/components/src/shared/WithEvents.js

```diff
@@ -68,7 +70,7 @@ export default (eventDefinitons, OriginalComponent) => {
     Derived.prototype[d.addMethod] = function(callback) {
       if (this[listeners][d.name].indexOf(callback === -1)) {
         this[listeners][d.name].push(callback)
-        this.lastEvt && callback(this.lastEvt)
```

Comment:
One mall small wrinkle. As this is a mix in, please don't pollute `this` as there name coalissions are easy. You can always keep this stateful variable at the function level, just below `const inChildContext`